src/Data/FunctorF.purs

```diff
@@ -0,0 +1,17 @@
+module Data.FunctorF where
+
+import Data.NaturalTransformation (type (~>))
+
+-- | A type class for higher order functors; types of kind `(* -> *) -> * -> *`.
```

Comment:
Actually, sorry, that's not right. It's "functors in the type constructor category" I suppose, since we don't require `Functor f`.